.cirrus.yml

```diff
@@ -0,0 +1,13 @@
+container:
+  image: golang:1.13
+
+env:
+  GOPROXY: https://proxy.golang.org
+
+validate_task:
+  validate_script:
+    - apt-get update -qq
+    - apt-get install -qq -y libgpgme-dev libdevmapper-dev btrfs-tools libbtrfs-dev
+    - make tools
+    - make .gitvalidation
+    - make validate
```

Comment:
Moving to Cirrus is a pretty big deal, not even mentioning it in the CI description is rather sneaky. Do I understand correctly that we now need _both_ systems to work to keep the current test coverage?

Can you explain how the move to Cirrus helps? Naively, both are Ubuntu-based containers with roughly similar configuration, aren’t they? (I’m sure the many iterations showed interesting or bizarre failures, but none of that knowledge is preserved in this PR.)

Especially the move of `.givalidation`, where, AFAICS, nothing changes about the _implementation_ of the test, and the previous Travis system seemed to work fine (didn’t it?), is rather curious.

(I’m not at all _insisting_ on Travis, it’s just that from the PR alone, it’s hard to tell whether there is really something wrong with Travis or whether all of this only makes a difference because it happens to paper over  some not-well-understood problem that is going to bite us even worse soon.)

Reply:
> Moving to Cirrus is a pretty big deal, not even mentioning it in the CI description is rather sneaky. Do I understand correctly that we now need _both_ systems to work to keep the current test coverage?

Yes, both need to work. I see using Cirrus as a _first_ step to become less dependent on Travis. Most containers/ projects are using Cirrus already and share a common GCP billing where we have _full_ control over the VM images. However, using Container tasks is for free in Cirrus, so I wanted to break the validation steps out into a dedicated task. I tried running golangci-lint in Travis but failed (with our setup) - more below.

> Can you explain how the move to Cirrus helps? Naively, both are Ubuntu-based containers with roughly similar configuration, aren’t they? (I’m sure the many iterations showed interesting or bizarre failures, but none of that knowledge is preserved in this PR.)

Some reasons are mentioned above. Both are container based and both run Ubuntu-based images, so there is no functional change yet. I could have very well also added another task to Travis but given the containers/ umbrella is all on Cirrus and that we can have full control over the VMs there, I see this as a _first step_ toward Cirrus.

> Especially the move of `.gitvalidation`, where, AFAICS, nothing changes about the _implementation_ of the test, and the previous Travis system seemed to work fine (didn’t it?), is rather curious.

The main driver for Cirrus was that I was unable to make golangci-lint run. The other PR didn't actually enable it as it silently failed. While trying to do that, I hit all kinds of errors in our setup which I attribute mostly to the (rather exotic?) way we run the container. Once I managed to setup a home dir, sudo fell apart and didn't manage to fix it. After a while I found the current setup to fragile to extend and decided to break out the validation step.

Ultimately, it's very easy to setup new (container-based) tasks in Cirrus that can run in parallel or in sequence with specified dependencies.

> (I’m not at all _insisting_ on Travis, it’s just that from the PR alone, it’s hard to tell whether there is really something wrong with Travis or whether all of this only makes a difference because it happens to paper over some not-well-understood problem that is going to bite us even worse soon.)

In a nutshell: I tried enabling golangci-lint but hit too many issues wiring that in our Travis setup. Breaking it out into a separate task made sense to me at some point; not only to have a clearer separation of CI tasks (e.g., validation, unit testing, integration testing) but also to not invest _too_ much time into this rather small linting adventure.

Reply:
> However, using Container tasks is for free in Cirrus, so I wanted to break the validation steps out into a dedicated task.

Thanks, that makes sense.

If the `.gitvalidation` step is for cleanliness, not to work around breakage (and now that there is a written record of it), ACK to the general concept of this PR. (I haven’t reviewed / studied the Cirrus setup it in detail.)